unchained-commons/src/main/java/unchained/commons/utils/ReflectionUtils.java

```diff
@@ -20,6 +22,310 @@ public static Type resolveCommonAncestor(List<Type> types) {
         return types.stream().reduce(ReflectionUtils::resolveCommonAncestor).orElse(null);
     }
 
+    public static <T, S extends T> Class<?> resolveRawArgument(Class<T> type, Class<S> subType) {
+        return resolveRawArgument(resolveGenericType(type, subType), subType);
+    }
+
+    public static Class<?> resolveRawArgument(Type genericType, Class<?> subType) {
+        Class<?>[] arguments = resolveRawArguments(genericType, subType);
+        if (arguments.length != 1) {
+            throw new IllegalArgumentException(
+                String.format("Expected 1 argument for generic type <%s> but found <%d>.", genericType, arguments.length));
+        }
+        return arguments[0];
+    }
+
+    public static <T, S extends T> Class<?>[] resolveRawArguments(Class<T> type, Class<S> subType) {
+        return resolveRawArguments(resolveGenericType(type, subType), subType);
+    }
+
+    public static Class<?>[] resolveRawArguments(Type genericType, Class<?> subType) {
+        Class<?>[] result = new Class<?>[0];
+        Class<?> functionalInterface = null;
```

Comment:
Having special purpose `null` values reduces code clarity. Instead, create overloads with default parameter values which are called in each case.